src/Shared/FileUtilities.cs

```diff
@@ -463,6 +463,10 @@ internal static bool LooksLikeUnixFilePath(string value)
             {
                 return true;
             }
+            else if (Directory.Exists(value) || File.Exists(value))
```

Comment:
FYI, this code is on the hot path (might get called hundreds of thousands of times on large builds). Though I don't see a way of reducing IO even further.